cmd/better-retester/retester.go

```diff
@@ -211,21 +211,46 @@ func (b *backoffCache) check(pr tide.PullRequest, baseSha string) (retestBackoff
 	return retestBackoffRetest, fmt.Sprintf("Remaining retests: %d against base HEAD %s and %d for PR HEAD %s in total", maxRetestsForShaAndBase-record.RetestsForBaseSha, record.BaseSha, maxRetestsForSha-record.RetestsForPrSha, record.PRSha)
 }
 
+func contains(elems []string, v string) bool {
+	for _, s := range elems {
+		if v == s {
+			return true
+		}
+	}
+	return false
+}
+
+func (c *retestController) createComment(pr tide.PullRequest, cmd, message string) {
+	comment := fmt.Sprintf("%s\n\n%s\n", cmd, message)
+	if err := c.ghClient.CreateComment(string(pr.Repository.Owner.Login), string(pr.Repository.Name), int(pr.Number), comment); err != nil {
+		c.logger.WithError(err).Error("failed to create a comment")
+	}
+}
+
 func (c *retestController) retestOrBackoff(pr tide.PullRequest) error {
 	branchRef := string(pr.BaseRef.Prefix) + string(pr.BaseRef.Name)
 	baseSha, err := c.ghClient.GetRef(string(pr.Repository.Owner.Login), string(pr.Repository.Name), strings.TrimPrefix(branchRef, "refs/"))
 	if err != nil {
 		return err
 	}
 
+	repo := fmt.Sprintf("%s/%s", pr.Repository.Owner.Login, pr.Repository.Name)
 	action, message := c.backoff.check(pr, baseSha)
 	switch action {
 	case retestBackoffHold:
-		c.logger.Infof("%s: %s (%s)", prUrl(pr), "/hold", message)
+		if contains(c.commentOnRepos, repo) {
```

Comment:
Let's make `commentOnRepos` a set, then it has a `Has` method and you do not need to reimplement. The `flagutil.Strings` even has a method to make the set for you:

https://github.com/kubernetes/test-infra/blob/7092a56c8e09bb0c23d782f7a165ed923fa4915c/prow/flagutil/strings.go#L52-L55

Make sure you only create the set once though - ideally the `retestController` has the set as a member, and you make it from `flagutil.Strings` when you make an instance of `retestControler`